core/io/resource_uid.cpp

```diff
@@ -45,23 +45,45 @@ String ResourceUID::get_cache_file() {
 	return ProjectSettings::get_singleton()->get_project_data_path().path_join("uid_cache.bin");
 }
 
+static constexpr uint8_t uuid_characters[] = { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', '0', '1', '2', '3', '4', '5', '6', '7', '8' };
+static constexpr uint32_t uuid_characters_element_count = (sizeof(uuid_characters) / sizeof(*(uuid_characters)));
+static constexpr uint8_t max_uuid_number_length = 19; // // Max 0x7FFFFFFFFFFFFFFF size is 19 digits
+
 String ResourceUID::id_to_text(ID p_id) const {
 	if (p_id < 0) {
 		return "uid://<invalid>";
 	}
+	// max_uuid_number_length + uid:// (6) + 1 for null.
 	String txt;
-
+	txt.resize(max_uuid_number_length + 7);
```

Comment:
Oh, I'm just realizing, if you define txt after `tmp_size` is known you can eliminate one resize call.

Reply:
Oh true, that's one call down. Thank you !